projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -306,6 +306,10 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
     }
     public set data(val: any[] | null) {
         this._data = (val) ? val : [];
+
+        if (this._data?.length && this.selectionService.get(this.id) && this.virtDir) {
+            this._value = this.createDisplayText(this.selection, []);
```

Comment:
We could move this in a `ngDoCheck` call for example which will be invoked after the `data` changes and is populated with some values, in which case we shouldn't need the check above.

Additionally, we may want to note somewhere in the docs that when a combo is bound to remote data and put in the grid, then depending on the delay for the requested data one can expect a few moments when the combo will remain blank without any items and without anything in its field. I think this is somewhat of a limitation since when enter edit mode we essentially recreate the cell component and respectively the combo, so it has to load everything from scratch.